Newsletters: add 309 (2024-06-28) (#1752)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2024-06-28-newsletter.md

---
title: 'Bitcoin Optech Newsletter #309'
permalink: /en/newsletters/2024/06/28/
name: 2024-06-28-newsletter
slug: 2024-06-28-newsletter
type: newsletter
layout: newsletter
lang: en
---
This week's newsletter FIXME

## News

- **Estimating the likelihood that an LN payment is feasible:** René
  Pickhardt [posted][pickhardt feasible1] to Delving Bitcoin about
  estimating the likelihood that an LN payment is feasible given the
  public knowledge of a channel's maximum capacity but no knowledge
  about its current balance distribution.  For example, Alice has a
  channel with Bob and Bob has a channel with Carol.  Alice knows the

>   capacity of the Bob-Carol channel but she doesn't know how much of

Reviewer: ```suggestion
  capacity of the Bob-Carol channel but doesn't know how much of
```

  that balance is controlled by Bob and how much is controlled by Carol.

  Pickhardt notes that some wealth distributions are impossible in a
  payment network.  For example, Carol can't receive more money in her
  channel with Bob than the capacity of that channel.  When all the
  impossible distributions are excluded, it can be useful to consider
  all the remaining wealth distributions as equally likely to occur.
  That can be used to produce a metric for the likelihood that a payment
  is feasible.

  For example, if Alice wants to send a 1 BTC payment to Carol, and the
  only channels it can pass through are Alice-Bob and Bob-Carol, then
  we can look at what percentages of wealth distributions in the
  Alice-Bob channel and the Bob-Carol channel would allow that payment

>   to succeed.  If the Alice-Bob channel has capacity of several BTC,

Reviewer: ```suggestion
  to succeed.  If the Alice-Bob channel has a capacity of several BTC,
```

  most possible wealth distributions would allow the payment to succeed.

>   If the Bob-Carol channel has a capacity just barely over 1 BTC, then

Reviewer: ```suggestion
  If the Bob-Carol channel has a capacity of just barely over 1 BTC, then
```

  most possible wealth distributions would prevent the payment from
  succeeding.  This can be used to calculate the overall likelihood
  of the feasibility of a payment of 1 BTC from Alice to Carol.

  The likelihood of feasibility makes it clear that many LN payments
  that naively seem possible will not succeed in practice.  It also
  provides a useful basis for comparison: Pickhardt provides a graph for
  a simulated network that compares the likelihood of feasibility to the

>   likelihood of a minimum cost flow solver making a successful payment,

Reviewer 2: Just to clarify: I only compare the likelihood of feasibility with the likelihood that an **attempt** that is taken from the MCF approach is successful. The later makes no statement about the likelihood weather the **payment** will be successful or feasible.

>   showing the solver succeeding most of the time a payment is possible.

Reviewer 2: I am not sure if I show this.

nswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

FIXME:bitschmidty

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

- [Bitcoin Core 26.2rc1][] is a release candidate for a maintenance
  version of Bitcoin Core's older release series.

<!-- FIXME:harding to update Thursday -->

## Notable code and documentation changes

_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
repo], and [BINANAs][binana repo]._

- [Bitcoin Core #29575][] net_processing: make any misbehavior trigger immediate discouragement

- [Bitcoin Core #28984][] Cluster size 2 package rbf

> - [Core Lightning #7388][] no longer allow creation of old

Reviewer: ```suggestion
- [Core Lightning #7388][] no longer allows the creation of old
```

Author: (Ignoring; this isn't text for the newsletter, it's from the CLN commit message.  I include some text for each unwritten PR so that @Gustavojfe can tell if I've typed the wrong PR number or something.)

  (experimental-only!) non-zero-fee anchor channels. [...] We still

>   support existing ones, though we were the only implementation which

Reviewer: ```suggestion
  support existing ones, though we were the only implementation that
```

Author: (Ignoring; this isn't text for the newsletter)

  provides a useful basis for making comparisons.
  In a [reply][pickhardt feasible2], Pickhardt describes how the
  likelihood metric could be used by wallets and business software to
  automatically make some intelligent decisions on behalf of its users.

## Selected Q&A from Bitcoin Stack Exchange

*[Bitcoin Stack Exchange][bitcoin.se] is one of the first places Optech
contributors look for answers to their questions---or when we have a
few spare moments to help curious or confused users.  In
this monthly feature, we highlight some of the top-voted questions and
answers posted since our last update.*

{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aa
nswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

FIXME:bitschmidty

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

- [LND v0.18.1-beta][] is a minor release with a fix for "an [issue][lnd
  #8862] that arises when handling an error after attempting to

>   broadcasting transactions if a btcd backend with an older version

Reviewer: This is within a quote so may not be fixable?
```suggestion
  broadcast transactions if a btcd backend with an older version
```

Author: Made the edit.  I think it's fine.  I don't like editing if it's the quoted author's style choice or if there's any risk of changing the meaning, but small edits to fix clear typos are ok.

- [Core Lightning #7388][] removes the ability to create non-zero-fee
  [anchor-style][topic anchor outputs] channels to conform to recent changes in
  the BOLT specification, while maintaining support for existing channels. Core
  Lightning was the only implementation to have this, and only in experimental
  mode. Other updates include rejecting `encrypted_recipient_data` containing
  both `scid` and `node`, adding LaTeX formatting to the onion specification,
  and other BOLT specification changes mentioned in Newsletters [#259][news259
  bolts] and [#305][news305 bolts].

- [LND #8734][] improves the payment route estimation abort process when a user
  interrupts the `lncli estimateroutefee` RPC command by making the payment loop
  aware of the client's streaming context. Previously, interrupting this command
  would cause the server to continue [payment probing][topic payment probes]
  routes unnecessarily. See Newsletter [#293][news293 routefee] for a previous
  reference to this command. 

- [LDK #3127][] implements non-strict forwarding to improve payment reliability,

>   allowing [Hash Time Locked Contract (HTLCs)][topic htlc] to be forwarded to a

Author: @Gustavojfe FYI: we don't need to expand abbreviations for which we have a topic page; it's ok to just link to them instead (e.g. `[HTLC][topic htlc]`).  TRUC, used above, is still a bit new, so I'm leaving that in, but I think we can stop expanding that in future newsletters.

- [What is `progress increase per hour` during synchronization?]({{bse}}123279)
  Pieter Wuille clarifies that progress increase per hour is the percentage of
  the blockchain synchronized per hour and not an increase in progress rate. He goes on
  to note reasons why progress is not constant and can vary.

- [Should an even Y coordinate be enforced after every key-tweak operation, or only at the end?]({{bse}}119485)
  Pieter Wuille agrees that when to perform a key negation to enforce [x-only
  public keys][topic X-only public keys] is largely a matter
  of opinion while pointing out pros and cons within different protocols.

- [Signet mobile phone wallets?]({{bse}}123045)
  Murch lists four [signet][topic signet]-compatible mobile wallet apps:
  Nunchuk, Lava, Envoy, and Xverse.

- [What block had the most transaction fees? Why?]({{bse}}7582)
  Murch uncovers block 409,008 with the most bitcoin-denominated fees (291.533
  BTC caused by a missing change output) and block 818,087 with the most
  USD-denominated fees ($3,189,221.5 USD, also presumed to be caused by a missing
  change output).

- [bitcoin-cli listtransactions fee amount is way off, why?]({{bse}}123391)

>   Ava Chow notes this discepency occurs when Bitcoin Core's wallet is unaware of

Reviewer 3: ```suggestion
  Ava Chow notes this discrepancy occurs when Bitcoin Core's wallet is unaware of
```

Reviewer 4: Thank you @azuchi
